R/RSocrata.R

```diff
@@ -262,6 +262,11 @@ read.socrata <- function(url, app_token = NULL, email = NULL, password = NULL,
 	validUrl <- validateUrl(url, app_token) # check url syntax, allow human-readable Socrata url
 	parsedUrl <- httr::parse_url(validUrl)
 	mimeType <- mime::guess_type(parsedUrl$path)
+	orderTest <- function (x) x == "$order"
+	if (!is.null(names(parsedUrl$query))) # check if URL has any queries 
+	  if(sum(sapply(names(parsedUrl$query), orderTest)) == 0) # check if URL is sorted
+	    # sort by Socrata unique identifier
+	    validUrl <- paste(validUrl, if(is.null(parsedUrl$query)) {'?'} else {"&"}, '$order=:id', sep='') 

```

Comment:
I like the readability here.

Although, as I'm saying that, can I suggest this for the previous lines?
```
    if (!is.null(names(parsedUrl$query))) # check if URL has any queries
        ## if there is a query, check for $order within the query
        orderTest <- any(names(parsedUrl$query == "$order"))
        if(orderTest)
            ...
```
Although I love the trick of summing logicals to mean "or", I'm pretty sure that will work without `sapply`.